### PR TITLE
Aidand tool calls

### PR DESCRIPTION
**Pull request by aidando73**
# What does this PR do?

In short, provide a summary of what this PR does and why. Usually, the relevant context should be present in a linked issue.

- [ ] Addresses issue (#issue)

## Test Plan

Please describe:
 - tests you ran to verify your changes with result summaries.
 - provide instructions so it can be reproduced.

## Sources

Please link relevant resources if necessary.

## Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Ran pre-commit to handle lint / formatting issues.
- [ ] Read the [contributor guideline](https://github.com/meta-llama/llama-stack/blob/main/CONTRIBUTING.md),
      Pull Request section?
- [ ] Updated relevant documentation.
- [ ] Wrote necessary unit or integration tests.

### CODE REVIEW

**Comment by aidando73** [comment]
tl;dr: The model is hallucinating and we don't check whether the client passed in the `code_interpreter` tool.

I've submitted a PR so that we at least don't get a 500 - but preventing it from hallucinating isn't too straightforward.

<details>
<summary>Explanation</summary>

The inference [request](https://github.com/meta-llama/llama-stack/blob/a05e7a583739353d7de6f9e87b0341bd9d8118fe/llama_stack/providers/remote/inference/ollama/ollama.py#L212) the model gets is correct:
```python
tools=[ToolDefinition(tool_name=<BuiltinTool.brave_search: 'brave_search'>, description=None, parameters=None)]
```
Doesn't include `code_interpreter`.

The raw message I get back is:

```python
<|python_tag|>import os
from llama_index import llama_recipes
from llama_index.finetuning import main


def fine_tune_lamaguard():
    # Define the path to the finetuning configuration file
...
```

The `<|python_tag|>` is interpreted as a [tool_call and defaults to code_interpreter](https://github.com/meta-llama/llama-models/blob/7890266c5a3ccd29e739d53a71ea968bcf4ca400/models/llama3/api/chat_format.py#L156-L194). But when we parse, we don't check whether the client passed in the code_interpreter tool:

https://github.com/meta-llama/llama-stack/blob/cde9bc1388fac4481b6032dc91bc0e2502e98734/llama_stack/providers/utils/inference/openai_compat.py#L230-L240

A simple fix would be to only return a tool_call if they're explicitly requested:

```python
    for tool_call in message.tool_calls:
        if tool_call.name in enabled_tools:
            yield ChatCompletionResponseStreamChunk(
                event=ChatCompletionResponseEvent(
                    event_type=ChatCompletionResponseEventType.progress,
                    delta=ToolCallDelta(
                        content=tool_call,
                        parse_status=ToolCallParseStatus.success,
                    ),
                    stop_reason=stop_reason,
                )
            )
```

This would return weird responses like `brave_search.call(query="NBA Western Conference Semifinals 2024 winning team roster")` if the user didn't specify brave_search as a tool call (and the model hallucinates) but that's probably more preferable to a 500.
</details>

**Comment by aidando73** [comment]
tl;dr: The model is hallucinating and we don't check whether the client passed in the `code_interpreter` tool.

I've submitted a PR so that it at least isn't a 500 - but preventing it from hallucinating isn't too straightforward.

### Explanation

The inference [request](https://github.com/meta-llama/llama-stack/blob/a05e7a583739353d7de6f9e87b0341bd9d8118fe/llama_stack/providers/remote/inference/ollama/ollama.py#L212) the model gets is correct:
```python
tools=[ToolDefinition(tool_name=<BuiltinTool.brave_search: 'brave_search'>, description=None, parameters=None)]
```
Doesn't include `code_interpreter`.

The raw message I get back is:

```python
<|python_tag|>import os
from llama_index import llama_recipes
from llama_index.finetuning import main


def fine_tune_lamaguard():
    # Define the path to the finetuning configuration file
...
```

The `<|python_tag|>` is interpreted as a [tool_call and defaults to code_interpreter](https://github.com/meta-llama/llama-models/blob/7890266c5a3ccd29e739d53a71ea968bcf4ca400/models/llama3/api/chat_format.py#L156-L194). But when we parse, we don't check whether the client passed in the code_interpreter tool:

https://github.com/meta-llama/llama-stack/blob/cde9bc1388fac4481b6032dc91bc0e2502e98734/llama_stack/providers/utils/inference/openai_compat.py#L230-L240

A simple fix would be to only return a tool_call if they're explicitly requested:

```python
    for tool_call in message.tool_calls:
        if tool_call.name in enabled_tools:
            yield ChatCompletionResponseStreamChunk(
                event=ChatCompletionResponseEvent(
                    event_type=ChatCompletionResponseEventType.progress,
                    delta=ToolCallDelta(
                        content=tool_call,
                        parse_status=ToolCallParseStatus.success,
                    ),
                    stop_reason=stop_reason,
                )
            )
```

This would return weird responses like `brave_search.call(query="NBA Western Conference Semifinals 2024 winning team roster")` if the user didn't specify brave_search as a tool call (and the model hallucinates) but that's probably more preferable to a 500.
